src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -1212,18 +1212,23 @@ public async Task<MonoBinaryReader> GetTypePropertiesReader(SessionId sessionId,
             if (typeInfo == null)
                 return null;
 
-            if (typeInfo.PropertiesBinaryReader != null)
+            if (typeInfo.PropertiesBuffer != null)
             {
-                typeInfo.PropertiesBinaryReader.BaseStream.Seek(0, SeekOrigin.Begin);
-                return typeInfo.PropertiesBinaryReader;
+                var retDebuggerCmd = new MemoryStream(typeInfo.PropertiesBuffer);
+                var retDebuggerCmdReader = new MonoBinaryReader(retDebuggerCmd);
+                return retDebuggerCmdReader;
             }
 
             var commandParams = new MemoryStream();
             var commandParamsWriter = new MonoBinaryWriter(commandParams);
             commandParamsWriter.Write(typeId);
 
-            typeInfo.PropertiesBinaryReader = await SendDebuggerAgentCommand<CmdType>(sessionId, CmdType.GetProperties, commandParams, token);
-            return typeInfo.PropertiesBinaryReader;
+            var reader = await SendDebuggerAgentCommand<CmdType>(sessionId, CmdType.GetProperties, commandParams, token);
+            var ms = new MemoryStream();
+            reader.BaseStream.CopyTo(ms);
+            typeInfo.PropertiesBuffer = ms.ToArray();
```

Comment:
Would it make sense to use `typeInfo.PropertiesBuffer = reader.ReadBytes(reader.BaseStream.Length);`?

Reply:
I think I need this cast to work: `typeInfo.PropertiesBuffer = reader.ReadBytes((int)reader.BaseStream.Length);`
Are you okay with that?

Reply:
done